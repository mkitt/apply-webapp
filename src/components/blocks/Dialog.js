// @flow
import React, { type Node } from 'react'
import glamorous from 'glamorous'
import Button from '../elements/Button'
import Heading from '../elements/Heading'
import { COLORS, MQ } from '../../constants/theme'

type Props = {
  cancelLabel?: string,
  children?: Node,
  confirmLabel?: string,
  handleCancel?: Function,
  handleConfirm?: Function,
  title?: string,
}

const Dialog = ({
  cancelLabel,
  children,
  confirmLabel,
  handleCancel,
  handleConfirm,
  title,
}: Props) => (
  <View>
    { title != null && <Heading>{title}</Heading> }
    { children != null && <Content>{children}</Content> }
    { (handleCancel || handleConfirm) &&
      <Actions>
        { handleCancel &&
          <Button box secondary onClick={handleCancel}>
            {cancelLabel}
          </Button>
        }
        { handleConfirm &&
          <Button box primary onClick={handleConfirm}>
            {confirmLabel}
          </Button>
        }
      </Actions>
    }
  </View>
)

Dialog.defaultProps = {
  cancelLabel: 'cancel',
  children: null,
  confirmLabel: 'confirm',
  handleCancel: null,
  handleConfirm: null,
  title: null,
}

export default Dialog

// -------------------------------------

const View = glamorous.div({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  paddingTop: 112,
  [MQ.MIN_BREAK_31]: {
    width: 496,
  },
})

const Content = glamorous.div({
  width: '100%',
  marginTop: 16,
  marginBottom: 32,
  fontSize: 14,
  color: COLORS.GREY_8,
  [MQ.MIN_BREAK_31]: {
    marginBottom: 64,
  },
})

const Actions = glamorous.div({
  display: 'grid',
  gridTemplateColumns: '1fr',
  gridGap: 16,
  [MQ.MIN_BREAK_31]: {
    gridTemplateColumns: '1fr 1fr',
  },
})
