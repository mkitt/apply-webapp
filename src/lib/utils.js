// @flow
import { __, addIndex, always, and, complement, either, gt, isEmpty, isNil, map } from 'ramda'
import { withProps } from 'recompose'

// -------------------------------------
// Environments
export const isDev = always(process.env.NODE_ENV === 'development')

// -------------------------------------
// Types
export const isNotNil = complement(isNil)
export const isNotEmpty = complement(isEmpty)
export const isEmptyOrNil = either(isNil, isEmpty)
export const isNotEmptyAndNotNil = and(isNotNil, isNotEmpty)

// -------------------------------------
// Relation
export const gt0 = gt(__, 0)

// -------------------------------------
// Lists
export const nool = always([])()
export const mapIndexed = addIndex(map)

// -------------------------------------
// Objects
export const noob = always({})()

// -------------------------------------
// Functions
export const noop = () => {}

export const log = withProps(props => (
  // eslint-disable-next-line no-console
  console.log(props.displayName || 'üëç', props) && props
))

export const warn = withProps(props => (
  // eslint-disable-next-line no-console
  console.warn(props.displayName || '‚ö†Ô∏è', props) && props
))
