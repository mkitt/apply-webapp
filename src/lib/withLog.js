import { withProps } from 'recompose'

export default withProps(props => (
  // eslint-disable-next-line
  console.log(props.logName || props.displayName || '👍', props) && props
))
