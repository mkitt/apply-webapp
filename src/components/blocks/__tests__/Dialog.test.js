import React from 'react'
import renderer from 'react-test-renderer'
import { noop } from '../../../lib/utils'
import Dialog from '../Dialog'

describe('<Dialog />', () => {
  it('shows the correct snapshot tree for the <Dialog> block', () => {
    const component = (
      <Dialog
        cancelLabel="Cancel Label"
        confirmLabel="Confirm Label"
        handleCancel={noop}
        handleConfirm={noop}
        title="Title"
      >
        Dialog Children
      </Dialog>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
