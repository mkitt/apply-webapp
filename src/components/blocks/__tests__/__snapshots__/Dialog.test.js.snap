// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> shows the correct snapshot tree for the <Dialog> block 1`] = `
.css-0,
[data-css-0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: 112px;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-1,
[data-css-1] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  font-size: 34px;
  font-weight: 200;
  color: #808080;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.css-2,
[data-css-2] {
  width: 100%;
  margin-top: 16px;
  margin-bottom: 32px;
  font-size: 14px;
  color: #808080;
}

.css-3,
[data-css-3] {
  display: grid;
  grid-template-columns: 1fr;
  grid-gap: 16px;
  -ms-grid-template-columns: 1fr;
  -ms-grid-gap: 16px;
}

.css-4,
[data-css-4] {
  text-align: left;
  min-width: 240px;
  height: 48px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 14px;
  text-transform: uppercase;
  color: #fff;
  background-color: #999;
}

.css-5,
[data-css-5] {
  text-align: left;
  min-width: 240px;
  height: 48px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 14px;
  text-transform: uppercase;
  color: #fff;
  background-color: #4d4d4d;
}

@media (min-width: 31em) {
  .css-0,
  [data-css-0] {
    width: 496px;
  }
}

@media (min-width: 31em) {
  .css-2,
  [data-css-2] {
    margin-bottom: 64px;
  }
}

@media (min-width: 31em) {
  .css-3,
  [data-css-3] {
    grid-template-columns: 1fr 1fr;
    -ms-grid-template-columns: 1fr 1fr;
  }
}

<div
  className="css-0"
>
  <h2
    className="css-1"
  >
    Title
  </h2>
  <div
    className="css-2"
  >
    Dialog Children
  </div>
  <div
    className="css-3"
  >
    <button
      className="css-4"
      onClick={[Function]}
      role="button"
    >
      Cancel Label
    </button>
    <button
      className="css-5"
      onClick={[Function]}
      role="button"
    >
      Confirm Label
    </button>
  </div>
</div>
`;
